### PR TITLE
Remove duplicate instrument name property from kafka event listener

### LINKED ISSUE #19342: Avoid duplicating StartLiveData's Instrument property in KafkaEventListener
`StartLiveData` has an `Instrument` property but it is not accessible to `KafkaEventListener` where it is needed to subscribe to the correct topics (data stream). The workaround for this was to have a second `InstrumentName` property in `KafkaEventListener`. Since Owen's changes in #19320 this can be cleaned up by overriding `setAlgorithm` in `KafkaEventListener`, something like this:
```
void KafkaEventListener::setAlgorithm(const API::IAlgorithm &callingAlgorithm) {
  this->updatePropertyValues(callingAlgorithm);
  // Get the instrument name from StartLiveData so we can sub to correct topics
  if (callingAlgorithm.existsProperty("Instrument")) {
    m_instrumentName = callingAlgorithm.getPropertyValue("Instrument");
  }
}
```

Don't forget to remove the `InstrumentName` property and handle the case in `KafkaEventListener::connect` that `m_instrumentName` wasn't set.

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Removes "InstrumentName" property from the KafkaEventListener. Since the changes in #19320 the algorithm can instead access the "Instrument" property of StartLiveData.

**To test:**
Open Live Data Interface with ISIS as your default facility and select ZOOM as the instrument. There should not be an "InstrumentName" field in the bottom left of the interface and the algorithm should start without error using the following options:
"Starting Time": "Start of Run"
"At Run Transition": "Stop"
"Output Workspace": "outWs"
Leaving other settings as default is fine.

Fixes #19342.

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/LiveData/inc/MantidLiveData/Kafka/KafkaEventListener.h [hunk 1]
```diff
@@ -38,8 +43,7 @@ class KafkaEventStreamDecoder;
  */
 class DLLExport KafkaEventListener : public API::LiveListener {
 public:
-  KafkaEventListener();
-  /// Destructor. Should handle termination of any socket connections.
+  KafkaEventListener() = default;
```
Comment (OwenArnold) [inline Framework/LiveData/inc/MantidLiveData/Kafka/KafkaEventListener.h]: Very minor point (choose to ignore), but do we even need this line? As there isn't any other constructor defined, the compiler will make this anyway.

File: Framework/LiveData/src/Kafka/KafkaEventListener.cpp [hunk 2]
```diff
@@ -13,21 +14,32 @@ namespace LiveData {
...
+    m_instrumentName = callingAlgorithm.getPropertyValue("Instrument");
+  } else {
+    g_log.error("KafkaEventListener requires Instrument property to be set in "
+                "calling algorithm");
+  }
 }
 
 /// @copydoc ILiveListener::connect
 bool KafkaEventListener::connect(const Poco::Net::SocketAddress &address) {
+  if (m_instrumentName.empty()) {
```
Comment (OwenArnold) [inline Framework/LiveData/src/Kafka/KafkaEventListener.cpp]: What happens if the instrument name doesn't match a topic?

**Review by OwenArnold — COMMENTED** [review]
Generally, looks ok to me. Few minor comments.

Is this particularly hard to add unit tests for?

File: Framework/LiveData/src/Kafka/KafkaEventListener.cpp (hunk 2 above)
Comment (matthew-d-jones) [reply Framework/LiveData/src/Kafka/KafkaEventListener.cpp]: This is checked by the consumer, an error is logged and the algorithm terminates.

**Comment by matthew-d-jones** [comment]
I don't think any of the listener algorithms are tested, probably because injecting mocks for the network interface at this point is difficult. However `KafkaEventListener` itself does very little, `KafkaEventStreamDecoder` does all the work and is well tested.

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**